File: 1pac/1.10-Arbol/LinkedList.js
//Linked List Jerarquizado de menor a mayor

function Node(value){
    this.value = value;
    this.next = null;
    this.children = null;
}

function LinkedList(){
    this.first = null;

    this.add = LinkedListAdd;
    this.getLength = LinkedListGetLength;
    this.atPosition = LinkedListAtPosition;

}

    function LinkedListAdd(value){

        if(!this.first){
            this.first = new Node(value);
            return true;
        }
        else{
            //si el primero es mayor
            var c = new Compare();
            current = this.first;

            if(c.compare(value,current.value) < 0){
                this.first = new Node(value);
                this.first.next = current;
                return true;
            }
            else if(c.compare(value,current.value) == 0){
                this.first = new Node(value);
                this.first.next = current.next;
                return true;
            }
            else{
                    while(current.next){
                        previous = current;
                        current = current.next;
                        if(c.compare(value,current.value) < 0){
                            previous.next = new Node(value);
                            previous.next.next = current;
                            return true;
                        }
                        else if(c.compare(value,current.value) == 0){
                            previous.next = new Node(value);
                            previous.next.next = current.next;
                            return true;
                        }
                    }           
                    current.next = new Node(value);
                    return true;
            }
        }     
    }

    function LinkedListGetLength(){
        current = this.first;
        length = 0;
    
        while(current.next){
            length++;
            current = current.next;
        }
        length++;
    
        return length;
    }
    
    function LinkedListAtPosition(position){
        currentNode = this.first;
        currentPosition = 0;
        
        if(position > this.getLength()){
            return null;
        }
        else{
            while(currentNode.next){
                if(currentPosition == position){
                    return currentNode;
                }
                else{
                    currentPosition++;
                    currentNode = currentNode.next;
                }
            }
            if(currentPosition == position){
                return currentNode;
            }
        }
    
    }

File: Proyecto/ProyectoExploradorDeDirectorios/LinkedList.py
from Compare import *
class Node:

    def __init__(self,value):
        self.value = value
        self.next = None
        self.children = LinkedList()
    
class LinkedList:

    def __init__(self):
        self.first = None

    def add(self, value):
        if(not self.first):
            self.first = Node(value)
            return True
        else:
            c = Compare()
            current = self.first

            if(c.compare(value,current.value) < 0):
                self.first = Node(value)
                self.first.next = current
                return True
            elif(c.compare(value,current.value) == 0):
                self.first = Node(value)
                self.first.next = current.next
                return True
            else:
                while(current.next):
                    previous = current
                    current = current.next
                    if(c.compare(value,current.value) < 0):
                        previous.next = Node(value)
                        previous.next.next = current
                        return True
                    elif(c.compare(value,current.value) == 0):
                        previous.next = Node(value)
                        previous.next.next = current.next
                        return True
                current.next = Node(value)
                return(True)

    def getNode(self, value):
        if(not self.first):
            return False
        else:
            current = self.first
            while(current.next):
                if(value == current.value):
                    return current
                current = current.next

            if(value == current.value):
                return current
            return False

    def toArray(self):
        current = self.first
        ar = []
        if(current):
            while(current.next):
                ar.append(current.value)
                current = current.next
            ar.append(current.value)

        return ar
    
    def removeForValue(self,valueToRmv):
        current=self.first
        if(current.value==valueToRmv):
            self.first=current.next
            return True
        while(current.next):
            before=current
            current=current.next
            if(current.value==valueToRmv):
                before.next=current.next
                return True
        return False

